mas-foundation/src/main/java/com/ca/mas/core/storage/storagesource/MASStorageSource.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2017 CA. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ *
+ */
+package com.ca.mas.core.storage.storagesource;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import com.ca.mas.core.security.EncryptionProvider;
+import com.ca.mas.core.storage.StorageActions;
+import com.ca.mas.foundation.MAS;
+import com.ca.mas.foundation.MASFoundationStrings;
+
+/**
+ * MASStorageSource is designed for developers to write, read, and delete String or byte[] data into
+ * the AccountManager or the SharedPreferences. Using AccountManager allows that multiple applications signed with the same key and using the same
+ * account name can share data.
+ *
+ * Note: the framework should be initialized prior to using any of StorageActions's CRUD operations.
+ * <p>
+ * Requires the android.permission.AUTHENTICATE_ACCOUNTS and
+ * android:name="android.permission.MANAGE_ACCOUNTS" permissions
+ * in your application's AndroidManifest.xml.
+ * <p>
+ * By default, the massharedauthenticator.xml file will be used to create the accounts,
+ * but can be overridden in AndroidManifest.xml with your own xml file for another account type.
+ */
+@SuppressWarnings({"MissingPermission"})
+public class MASStorageSource {
```

Comment:
We may need to keep the original name MASSharedStorage, the MASShareStorage align with a class in iOS.
We can create a constructor with shared by default.